xdsl/ir/core.py

```diff
@@ -756,6 +756,14 @@ def walk(self) -> Iterator[Operation]:
         for region in self.regions:
             yield from region.walk()
 
+    def walk_reverse(self) -> Iterator[Operation]:
+        """
+        Iterate all operations contained in the operation (including this one) in reverse order.
+        """
+        for region in self.regions:
```

Comment:
My expectation of walk_reverse would be that `list(x.walk())[::-1] == list(x.walk_reverse())`, so this line would be something like this:
```suggestion
        for region in self.regions[::-1]:
```
Does that match your understanding of `walk_reverse()`? In either case, it would be nice to add a test with nested blocks and regions.

Reply:
I added the suggested test cases.